src/global_properties.rs

```diff
@@ -0,0 +1,194 @@
+        IxaError::JsonError(error)
+    }
+}
+
+impl std::error::Error for IxaError {}
+
+impl Display for IxaError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Error: {self:?}")?;
+        Ok(())
+    }
+}
+
+/// Global properties are not mutable and represent variables that are required
+/// in a global scope during the simulation, such as simulation parameters.
+pub trait GlobalProperty: Any {
+    type Value: Any;
+}
+
+pub use define_global_property;
+
+struct GlobalPropertiesDataContainer {
+    global_property_container: HashMap<TypeId, Box<dyn Any>>,
+}
+
+crate::context::define_data_plugin!(
+    GlobalPropertiesPlugin,
+    GlobalPropertiesDataContainer,
+    GlobalPropertiesDataContainer {
+        global_property_container: HashMap::default(),
+    }
+);
+
+pub trait ContextGlobalPropertiesExt {
+    fn set_global_property_value<T: GlobalProperty + 'static>(
+        &mut self,
+        property: T,
+        value: T::Value,
+    );
+    fn get_global_property_value<T: GlobalProperty + 'static>(&self, _property: T) -> &T::Value;
+
+    /// # Errors
+    ///
+    /// Will return an `IxaError` if the `file_path` does not exist or
+    /// cannot be deserialized
+    fn load_parameters_from_json<T: 'static + Debug + DeserializeOwned>(
+        &mut self,
+        file_path: &Path,
+    ) -> Result<T, IxaError>;
+}
+
+impl GlobalPropertiesDataContainer {
+    fn set_global_property_value<T: GlobalProperty + 'static>(
+        &mut self,
+        _property: &T,
+        value: T::Value,
+    ) {
+        let _data_container = self
+            .global_property_container
+            .entry(TypeId::of::<T>())
+            .or_insert_with(|| Box::new(value));
+    }
+
+    fn get_global_property_value<T: GlobalProperty + 'static>(&self) -> &T::Value {
+        let data_container = self
+            .global_property_container
+            .get(&TypeId::of::<T>())
+            .expect("Global property not initialized");
+        data_container.downcast_ref::<T::Value>().unwrap()
+    }
+}
+
+impl ContextGlobalPropertiesExt for Context {
+    fn set_global_property_value<T: GlobalProperty + 'static>(
+        &mut self,
+        property: T,
+        value: T::Value,
+    ) {
+        let data_container = self.get_data_container_mut(GlobalPropertiesPlugin);
+        data_container.set_global_property_value(&property, value);
+    }
+
+    #[allow(unused_variables)]
+    fn get_global_property_value<T: GlobalProperty + 'static>(&self, _property: T) -> &T::Value {
+        let data_container = self.get_data_container(GlobalPropertiesPlugin).unwrap();
+        data_container.get_global_property_value::<T>()
+    }
+
+    fn load_parameters_from_json<T: 'static + Debug + DeserializeOwned>(
+        &mut self,
+        file_name: &Path,
+    ) -> Result<T, IxaError> {
+        let config_file = fs::read_to_string(file_name)?;
+        let config = serde_json::from_str(&config_file)?;
```

Comment:
I'd probably open the file and use `from_reader` here https://docs.rs/serde_json/latest/serde_json/fn.from_reader.html rather than loading the whole file into a string first